packages/plugin-ext/src/main/browser/webview/webview-frontend-module.ts

```diff
@@ -0,0 +1,23 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { ContainerModule } from 'inversify';
+import { WidgetFactory } from '@theia/core/lib/browser';
+import { WebviewWidgetFactory } from './webview-widget-factory';
+
+export default new ContainerModule(bind => {
+    bind(WidgetFactory).toDynamicValue(ctx => new WebviewWidgetFactory(ctx.container)).inSingletonScope();
```

Comment:
I don’t think that we need many entry points. First because the widget cannot be created without factory, so code must stay together, otherwise now none of these entry points can work independent. Secondly there should be reuse reason for it and there is no such. Maybe in future we extract the webview extension and then it will have own entry point, but it will be designed properly then.

Reply:
> otherwise now none of these entry points can work independent.

Yes, but on the other hand I never saw entry points working independently within a given Theia extension. What I mean is that all modules defined under `theiaExtensions` get included.

> Secondly there should be reuse reason for it and there is no such.

The reason is that on the browser we want implementation X, but on Electron we want Y. The only mechanism I knew to have the proper implementation was to make mutually exclusive containers.

How would you do? This seemed like the simplest way of doing it somewhat clean.

Reply:
> Yes, but on the other hand I never saw entry points working independently within a given Theia extension. What I mean is that all modules defined under theiaExtensions get included.

It does not make it better to separate code which should run together to different entry points in ad-hoc manner. I don't seen the motivation here and it harms future readers since they won't find factory next to widget binding.

Reply:
Ok, I tried something else.